src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using CommunityToolkit.Maui.SourceGenerators.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Generators;
+
+[Generator]
+class TextColorToGenerator : ISourceGenerator
+{
+	public void Execute(GeneratorExecutionContext context)
+	{
+		var textStyleClassNameList = new List<string>();
+
+		var mauiControlsAssemblySymbol = context.Compilation.SourceModule.ReferencedAssemblySymbols.Single(q => q.Name == "Microsoft.Maui.Controls");
+
+		foreach (var namedTypeSymbol in mauiControlsAssemblySymbol.GlobalNamespace.GetNamedTypeSymbols())
+		{
+			if (namedTypeSymbol.ImplementsInterfaceOrBaseClass("ITextStyle"))
```

Comment:
Looks like this just scans the .NET MAUI source, that way this will ignore if the user creates his own control that implements the `ITextStyle` interface.

Reply:
Dang - I didn't even think about user-generated `ITextStyle` controls.

How do we handle that?

Reply:
we will need to implement a `SyntaxReceiver`, that way we will listen every time that users do some changes in the code... So to improve perf. we can try (I really don't know if that will be possible to do) to move this implementation to the `Initialize` method, this method is executed just once and the Maui types that implement `ITextStyle` will not change.
This will improve perf.

Then, on the `Execute` method we take about the user controls that implements `ITextStyle`.

Reply:
I can take a look at that with you next week. Anyway, if you want to play with it on your own [here is a reference for you](https://github.com/pictos/SourceGenerators/blob/main/StructGenerator/StructGenerator/StructGen.cs)

This is a source generator that I did that implements the `IEquatable<T>` interface on structs. So the only thing that the user needs to do to trigger the SG is to create a `partial struct` and implement `IEquatable<T>`. I can see that is the same thing that we want here :D

Reply:
I've also seen `IIncrementalGenerator` which is supposed to improve performance. 

https://andrewlock.net/exploring-dotnet-6-part-9-source-generator-updates-incremental-generators/

Reply:
I didn't try them yet. But yeah, that is more performant than normal `SourceGenerators`. We can try to implement it here